editor/components/post-permalink/index.js

```diff
@@ -72,8 +168,14 @@ export default connect(
 	( state ) => {
 		return {
 			isNew: isEditedPostNew( state ),
-			link: getEditedPostAttribute( state, 'link' ),
+			previewLink: getEditedPostPreviewLink( state ),
+			samplePermalinkData: getEditedPostAttribute( state, 'sample_permalink' ),
+
+			permalinkStructure: window.wpApiSettings.schema.permalink_structure,
```

Comment:
A few issues:

- As much as possible, we shouldn't access the `window` global directly, since it proliferates the expected global prerequisites beyond that which we're otherwise establishing through the [single entry point](https://github.com/WordPress/gutenberg/blob/b616d710d46516c5b252754b33d661ccb6964161/edit-post/index.js#L71) for initializing an editor
- This value is not being sourced from the editor state, and as such does not belong in [`connect`](https://github.com/reactjs/react-redux/blob/master/docs/api.md#connectmapstatetoprops-mapdispatchtoprops-mergeprops-options)'s `mapStateToProps` (not mapped from state)
- To the previous point, perhaps this _should_ be in editor state, or otherwise provided through context in [`EditorProvider`](https://github.com/WordPress/gutenberg/blob/master/editor/components/provider/index.js), where other settings are exposed and later accessed by [`withContext`](https://github.com/WordPress/gutenberg/tree/master/components/higher-order/with-context)
- To _this_ previous point, technically there is context already available for this provided through [`withAPIData`](https://github.com/WordPress/gutenberg/blob/master/components/higher-order/with-api-data/provider.js), but since we're not really using API data here, it doesn't seem necessarily appropriate to create a dependency here on this context.

Reply:
Turns out I didn't need that at all, everything was in `sample_permalink` on the post.